
* 2000-04-19 Jacek Konieczny (jajcus@pld.org.pl) 

This patch sets "SO_REUSEADDR" flag to server's listening socket.
This makes restart of the server possible after a crash.

diff -durN bzflag.orig/src/bzfs/bzfs.cxx bzflag/src/bzfs/bzfs.cxx
--- bzflag.orig/src/bzfs/bzfs.cxx	Tue Jan 25 16:31:15 2000
+++ bzflag/src/bzfs/bzfs.cxx	Wed Apr 19 11:43:20 2000
@@ -851,6 +851,7 @@
 
 static boolean		serverStart()
 {
+  int val;
   maxFileDescriptor = 0;
 
   // init addr:port structure
@@ -876,6 +877,8 @@
     nerror("couldn't make connect socket");
     return False;
   }
+  val=1;
+  setsockopt(wksSocket,SOL_SOCKET,SO_REUSEADDR,&val,sizeof(val));
   if (bind(wksSocket, (const struct sockaddr*)&addr, sizeof(addr)) == -1) {
     if (!useFallbackPort) {
       nerror("couldn't bind connect socket");
